libraries/recorder.js

```diff
@@ -117,16 +136,27 @@ recorder.persistToFile = function () {
     });
 };
 
+/**
+ * @return {string|null} log file name
+ */
 recorder.persistToFileSync = function() {
+    if (Object.keys(recorder.timeObject).length <= 1 && hasPersistedToFile) {
+        return null;
+    }
+
+    hasPersistedToFile = true;
+
     let logName = recorder.getCurrentLogName();
     let timeObjectStr = JSON.stringify(recorder.timeObject);
-    recorder.objectOld = {};
-    recorder.timeObject = {};
+    recorder.timeObject = {
+        [Date.now()]: JSON.parse(JSON.stringify(recorder.objectOld)),
```

Comment:
What if someone starts to use saveState(time) with some time param 'not aligned' with current time?
Maybe Date.now() could be replaced by the last time stamp in the timeObject which is going to be saved.

Reply:
That feels suitably elegant, modified accordingly